src/Core/LanguageService.fs

```diff
@@ -115,14 +115,93 @@ module LanguageService =
         else
             (fn + ".fsx")
 
+
+    let compilerLocation () =
+        match client with
+        | None -> Promise.empty
+        | Some cl ->
+            cl.sendRequest ("fsharp/compilerLocation", null)
+            |> Promise.map (fun (res: Types.PlainNotification) ->
+                let r = res.content |> ofJson<CompilerLocationResult>
+                r)
+
+    let tryFindDotnet () =
+        promise {
+            // User has specified a custom dotnet location
+            match Configuration.tryGet "FSharp.dotnetRoot" with
+            | Some dotnetRoot ->
+                // Choose the right program name depending on the OS
+                let program =
+                    if Environment.isWin then "dotnet.exe" else "dotnet"
+
+                // Compute the full path
+                let dotnetFullPath =
+                    path.join(dotnetRoot, program)
+
+                // Check if the program exists at the computed location
+                if fs.existsSync (U2.Case1 dotnetFullPath) then
+                    return Ok dotnetFullPath
+
+                else
+                    let msg =
+                        // The special syntax: %s{"\n" + dotnetFullPath}
+                        // Force a new line to be added, if I put %s{dotnetFullPath}
+                        // at the beginning of the next line, there is a compiler error...
+                        $"""
+Could not find `dotnet` in the `dotnetRoot` directory: %s{"\n" + dotnetFullPath}
+
+Consider:
+- updating the `FSharp.dotnetRoot` settings key to a directory with a `dotnet` binary
+- removing the `FSharp.dotnetRoot` settings key and :
+    - including `dotnet` in your PATH
+    - installing .NET Core"""
+
+                    return Core.Error msg
+
+            // No custom location was specified, try to find it from the PATH
+            | None ->
+                match! Environment.tryGetTool "dotnet" with
+                | Some dotnetFullPath ->
+                    return Ok dotnetFullPath
+
+                | None ->
+                    let msg = """Could not find `dotnet` in the path.
+
+Consider:
+- setting the `FSharp.dotnetRoot` settings key to a directory with a `dotnet` binary
+- including `dotnet` in your PATH
+- installing .NET Core
+"""
+
+                    return Core.Error msg
+        }
+
+// Old implementation: How does compilerLocation work?
+// If dotnet is not found on the system, client cannot be run, no?
```

Comment:
excellent question :D compilerlocation is something that existed in the old FSAC STDIO implementation that worked for non-dotnet-SDK builds. Since we've dropped those we can kill compilerlocation entirely, both here and in FSAC.

Reply:
Ok, I removed that code.